pkg/credentials/provider_test.go

```diff
@@ -47,11 +52,41 @@ func TestNewProviderAssignsSessionDuration(t *testing.T) {
 
 	stsProvider := StaticSTSProvider(nil)
 
-	provider, err := NewProvider(audience, roleArn, jwtSource, nonZeroSessionDuration, stsProvider)
+	provider, err := NewProvider(
+		audience,
+		roleArn,
+		jwtSource,
+		nonZeroSessionDuration,
+		stsProvider,
+		defaultEndpoint)
```

Comment:
I "rotated" the method arguments so that the lines would be shorter and easier to read. defaultEndpoint is new.